integration-tests/src/test/java/oracle/kubernetes/operator/ItMiiConfigUpdateCm.java

```diff
@@ -0,0 +1,366 @@
+
+  /**
+   * This method does the initialization of the integration test
+   * properties defined in OperatorIT.properties and setting
+   * the resultRoot, pvRoot and projectRoot attributes.
+   *
+   * @throws Exception if initializing the application properties
+   *         and creates directories for results fails.
+   */
+  @BeforeAll
+  public static void staticPrepare() throws Exception {
+    namespaceList = new StringBuffer();
+    testClassName = new Object() {
+    }.getClass().getEnclosingClass().getSimpleName();
+    // initialize test properties and create the directories
+    initialize(APP_PROPS_FILE, testClassName);
+  }
+
+  /**
+   * This method creates the result/pv root directories for the test.
+   * Creates the operator if its not running.
+   *
+   * @throws Exception if Operator creation fails
+   */
+  @BeforeEach
+  public void prepare() throws Exception {
+    createResultAndPvDirs(testClassName);
+
+    // create operator
+    if (operator == null) {
+      Map<String, Object> operatorMap = createOperatorMap(getNewSuffixCount(),
+          true, testClassName);
+      operator = TestUtils.createOperator(operatorMap, RestCertType.SELF_SIGNED);
+      assertNotNull(operator);
+      domainNS = ((ArrayList<String>) operatorMap.get("domainNamespaces")).get(0);
+      namespaceList.append((String)operatorMap.get("namespace"));
+      namespaceList.append(" ").append(domainNS);
+    }
+  }
+
+  /**
+   * Releases k8s cluster lease, archives result, pv directories.
+   *
+   * @throws Exception when errors while running statedump.sh or cleanup.sh
+   *         scripts or while renewing the lease for shared cluster run
+   */
+  @AfterAll
+  public static void staticUnPrepare() throws Exception {
+    tearDown(new Object() {
+    }.getClass().getEnclosingClass().getSimpleName(), namespaceList.toString());
+
+    LoggerHelper.getLocal().info("SUCCESS");
+  }
+
+  /**
+   * Create a domain without a JDBC DS using model in image and having config map
+   * in the domain.yaml before deploying the domain. After the domain resources
+   * is created, re-create the config map with model files that define a JDBC DataSource
+   * and patch to change domain-level restart version to reload the model files,
+   * generate new config and initiate a rolling restart.
+   */
+  @Test
+  public void testMiiConfigUpdateNonJdbcCm() {
+    Assumptions.assumeTrue(QUICKTEST);
+    String testMethodName = new Object() {
+    }.getClass().getEnclosingMethod().getName();
+    boolean testCompletedSuccessfully = false;
+    Domain domain = null;
+
+    try {
+      logTestBegin(testMethodName);
+      LoggerHelper.getLocal().log(Level.INFO,
+          "Creating Domain & waiting for the script to complete execution");
+
+      // create domain w/o JDBC DS using the image created by MII
+      boolean createDS = false;
+      domain = createDomainUsingMii(createDS, domainNS, testClassName);
+      assertNotNull(domain, "Failed to create a domain");
```

Comment:
I'm not sure for the created domain  assertNotNull would be enough. How about domain.verifyDomainCreated() ?

Reply:
domain.verifyDomainCreated()  is called in MiiBaseTest.java, createMiiDomainWithConfigMap. assertNotNull is just a second layer of check. I know it makes it harder to find if we have layers of extended class.